FC-1100  Permissions System (#1)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/permission-system.md

indicator: both
---

There are 3 user permissions pertaining to Functions:

* FaaS-Admin - allowed to read `lambdas`, manage the whitelist for external domains and manage secrets. This permission is by default set to 'ON' for all admins on the account. It is however possible to limit your admin's access as well. You can create a separate permission group for developers you wish to have part of the admin profile permissions and assign them with the relevant permissions as needed, by creating a custom profile.

* FaaS-Developer - allowed to manage `lambdas`, read whitelisted domains and read encrypted secrets.

> * FaaS-Invocation - only allowed to invoke lambdas externally. This permission is granted to a user dedicated to external invocations. When an external system needs to invoke a Functions `lambda`, the external system needs to use an account user with this permission <strong>enabled</strong> to access the LivePerson platform.

Reviewer: Don't need to go into greater detail. People will learn what they need, under external invocations.

<table class="thinner" style="width: 100%">
<thead>
  <tr>
    <th>Permission name	</th>
    <th  style="text-align: center; vertical-align: middle;">Agent</th>
    <th  style="text-align: center; vertical-align: middle;">Agent Manager</th>
    <th  style="text-align: center; vertical-align: middle;" >Campaign Manager</th>
    <th  style="text-align: center; vertical-align: middle;">Admin</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>FaaS-Admin</td>
    <td style="text-align: center; vertical-align: middle;" >❌</td>
    <td style="text-align: center; vertical-align: middle;" >🔒	</td>
    <td style="text-align: center; vertical-align: middle;" >❌</td>
    <td style="text-align: center; vertical-align: middle;" >✅</td>
  </tr>
    <tr>
    <td>FaaS-Developer</td>
    <td style="text-align: center; vertical-align: middle;" >❌</td>
    <td style="text-align: center; vertical-align: middle;" >🔒</td>
    <td style="text-align: center; vertical-align: middle;" >❌</td>
    <td style="text-align: center; vertical-align: middle;" >✅</td>
  </tr>
    <tr>
    <td>FaaS-Invocation</td>

>     <td style="text-align: center; vertical-align: middle;" >🔒</td>

Reviewer: Maybe we need to add a s short Legend under the tables

    <td>create/edit/delete</td>
    <td>✅</td>
    <td>✅</td>
    <td>-</td>
  </tr>
  <tr>
    <td>emailReport settings</td>
    <td>create/edit/delete</td>
    <td>✅</td>
    <td>-</td>
    <td>-</td>
  </tr>
  <tr>
    <td>emailReport recipients</td>
    <td>create/delete</td>
    <td>✅</td>
    <td>✅</td>
    <td>-</td>
  </tr>
  <tr>
    <td>emailReport</td>
    <td>read</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
   <tr>
    <td>logs</td>
    <td>read</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
  <tr>
    <td>logs</td>
    <td>create</td>
    <td>✅</td>
    <td>-</td>
    <td>-</td>
  </tr>
  <tr>
    <td>reporting</td>
    <td>read</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
</tbody>
</table>

You can configure these permissions in Conversational Cloud. More info on adding permissions can be found [here](https://knowledge.liveperson.com/admin-settings-permissions-profiles.html),the process is simple:

* Click on the users tab at the top of the Conversational Cloud UI and then click profiles.

* Create/Edit a profile and add the relevant FaaS-* permission to it.

![FaaSPermissionSystem](img/faas-permission-system.png)

### Hints

> * Be aware that user roles like Agent won't have the option to get FaaS-Developer and FaaS-Admin permission, you would need to use a different user with the proper role.

Reviewer: ```suggestion
* Please be aware that roles like Agent don't have the option to get FaaS Developer/Admin permission. You will need to use a different role in that case.
```

Author: actually is should be a profile Maybe the last sentence could be: _You will need to use another profile with a different role in that case_

> * CLI users are following the same rules. If you create a user to be used in a CI development it requires the same permissions.

Reviewer: Maybe you want to link `CLI` directly to the CLI Page

```suggestion
* The CLI leverages the same permissions, therefore you need to make sure that the user used has at least the FaaS-Developer permission.
```

> * Use only the minimum permission required. E.g, for external invocation use only a user with FaaS-Invocation permission.

Reviewer: ```suggestion
* We suggest obeying the security principle of least privilege and therefore only selecting the minimal set of permissions. E.g. If the user will be only using function (invoking). Then the FaaS-Invocation is only required.
```

* FaaS-Developer - allowed to manage `lambdas`, read whitelisted domains and read encrypted secrets.

> * FaaS-Invocation -provides basic read access on all functions and permission to invoke them. The Agent profile has this permission enabled by default.

Reviewer: ```suggestion
* FaaS-Invocation - provides basic read access on all functions and permission to invoke them. The Agent profile has this permission enabled by default.
```
